### PR TITLE
Keychron K3 ISO OS selector, Mac keymap and fixes

### LINKED ISSUE #177: [Feature Request] Support Caps lock LED on Keychron K3
The red Caps lock indicator LED does not work on the Keychron K3 (v1, Optical RGB). Instead of turning on/off when capslock is pressed, it is always dimly on.

## Feature Request Type

- [ ] Core functionality
- [ ] Add-on hardware support (eg. audio, RGB, OLED screen, etc.)
- [x] Alteration (enhancement/optimization) of existing feature(s)
- [ ] New behavior

## Description

The build for the RGB Keychron K3 with optical switches works if the `OPTICAL_MATRIX` macro is defined, but the Caps lock indicator feature does not seem to be working. The LED is expected to turn on if Caps lock is enabled and off if Caps lock is disabled. Instead,  the LED is always in a dim-on state, which could indicate a floating gate condition.

PCB version: Ver 1.4/20201030

Built from 03c781d3ca5b024ec3c21d85ad5f323aff70d842 with arm-none-eabi-gcc version 10.1.0 on Windows 10 with QMK MSYS 1.5.2.

### PR DESCRIPTION
**Pull request by Gibbio**
Added MAC/iOS Keyboad map and fixed OS selector switch
Fixed #177
Added missing special key for Windows and Mac:
KC_TASK (Task viewer) Fn + F3
KC_FLXP (Windows file explorer) Fn + F4
KC_SNIP (Windows snip tool) Fn + Print Screen
KC_MSSN (Mission Control) Fn + F3
KC_FIND (Finder) Fn + F4
KC_MSCR (Mac screenshot) Print Screen
KC_MSNP (Mac snip tool) Fn + Print Screen
KC_BRIU (Brightness Up) Fn + F1
KC_BRID (Brightness Down) Fn + F2

## Description
Keychron K3 ISO OS selector, Mac keymap and fixes

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/k3/keymaps/iso/keymap.c [hunk 1]
```diff
@@ -21,13 +21,25 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // Layer names don't all need to be of the same length, obviously, and you can also skip them
 // entirely and just use numbers.
 enum layer_names {
-  _BASE = 0,
-  _FN,
+  MAC_BASE = 0,
+  WIN_BASE,
+  MAC_FM,
```
Comment (dexter93) [inline keyboards/keychron/k3/keymaps/iso/keymap.c]: _FM is a weird choice for a FN layer. Any specific reason why?

File: keyboards/keychron/k3/keymaps/iso/keymap.c [hunk 2]
```diff
@@ -21,13 +21,25 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 
+#define KC_TASK LGUI(KC_TAB)        // Task viewer
+#define KC_FLXP LGUI(KC_E)          // Windows file explorer
+#define KC_SNIP LGUI(LSFT(KC_S))    // Windows snip tool
+
+#define KC_MSSN LGUI(KC_F3)         // Mission Control
+#define KC_FIND LALT(LGUI(KC_SPC))  // Finder
+#define KC_MSCR LSFT(LGUI(KC_3))    // Mac screenshot
+#define KC_MSNP LSFT(LGUI(KC_4))    // Mac snip tool
+
```
Comment (dexter93) [inline keyboards/keychron/k3/keymaps/iso/keymap.c:38]: Copying keychron Q1 here? Nice

Comment (Gibbio) [reply keyboards/keychron/k3/keymaps/iso/keymap.c:38]: Yes, I was using process_record_user, host_consumer_send(...) for mission control and launchpad but then i saw that Q1 already handle those keys and I just copy/paste the same defs.

File: keyboards/keychron/k3/keymaps/iso/keymap.c (hunk 1 above)
Comment (Gibbio) [reply keyboards/keychron/k3/keymaps/iso/keymap.c]: urgh, just a typo, should be MAC_FN and WIN_FN

**Pull request merged**